Objects/descrobject.c

```diff
@@ -1535,10 +1537,30 @@ property_deleter(PyObject *self, PyObject *deleter)
 }
 
 
+PyDoc_STRVAR(set_name_doc,
+             "Method to set name of a property.");
+
+static PyObject *
+property_set_name(PyObject *self, PyObject *args) {
+    propertyobject *prop = (propertyobject *)self;
+    PyObject *name = PyTuple_GetItem(args, 1);
+
+    if (name == NULL) {
+        return NULL;
+    }
+
+    Py_XINCREF(name);
+    Py_XSETREF(prop->prop_name, name);
+
+    Py_RETURN_NONE;
+}
+
+
 static PyMethodDef property_methods[] = {
     {"getter", property_getter, METH_O, getter_doc},
     {"setter", property_setter, METH_O, setter_doc},
     {"deleter", property_deleter, METH_O, deleter_doc},
+    {"__set_name__", property_set_name, METH_VARARGS, set_name_doc},
```

Comment:
Is there any reason this can't be `METH_O`?

Reply:
`__set_name__`  method accepts 2 args `owner` and `name`, so as far as I understand it can't be used with `METH_O` flag because it requires a function to has only one argument.

Please, correct me if I am wrong.